feat:#719 - prompts entries and ads drafts

marclupanc opened the pull request:
### 🛠 Description
Fixes #719 

This PR implements saving of Prompts, Entries or Advertisements in case submission process fails and creation modal closes. A draft copy is being saved to local storage and on next try fields are populated automatically with offering user the option to reset the fields and start over. 

This also implements automatic image removal from DB in case creation process fails. 

To note that images are not being saved in local storage as drafts. 

### ✨ Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### 🧠 Rationale behind the change

Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🧪 All Test Suites Passed?

- [X] Manual tested
- [ ] Cypress

marclupanc committed:
feat:#719 - prompts entries and ads drafts

marclupanc committed:
feat:#719 - remove consoles

sana7044 reviewed (changes requested):
@marclupanc 
After reviewing this PR, I have identified several issues that need to be addressed:

Draft Saving: The draft saving feature does not seem to work as intended. The fields are not being populated automatically on subsequent attempts after a failure.

Local Storage Handling: There may be a need for clarification or adjustment in how local storage is being managed, particularly regarding images not being saved as drafts.

The entry creation prompt selecting time, but it shows 'undefined' and does not create an entry. 

Please resolve the conflict.
### Issues
![Screenshot from 2025-03-22 00-18-25](https://github.com/user-attachments/assets/40fc8fd0-eca2-431f-bc31-447634052177)

https://www.awesomescreenshot.com/video/37901107?key=df9314644f6685c36d16a680631c17b2

marclupanc committed:
feat:#719 - fix code review

marclupanc committed:
feat:#719 - remove ad from localstorage if created

marclupanc committed:
feat:#719 - remove draft image for ads

marclupanc committed:
Merge branch 'develop' into 719-failing-adds-prompts-entries-should-be-saved-as-drafts

marclupanc commented:
> @marclupanc After reviewing this PR, I have identified several issues that need to be addressed:
> 
> Draft Saving: The draft saving feature does not seem to work as intended. The fields are not being populated automatically on subsequent attempts after a failure.
> 
> Local Storage Handling: There may be a need for clarification or adjustment in how local storage is being managed, particularly regarding images not being saved as drafts.
> 
> The entry creation prompt selecting time, but it shows 'undefined' and does not create an entry.
> 
> Please resolve the conflict.
> 
> ### Issues
> ![Screenshot from 2025-03-22 00-18-25](https://private-user-images.githubusercontent.com/78159045/425617979-40fc8fd0-eca2-431f-bc31-447634052177.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI2NDU1NjAsIm5iZiI6MTc0MjY0NTI2MCwicGF0aCI6Ii83ODE1OTA0NS80MjU2MTc5NzktNDBmYzhmZDAtZWNhMi00MzFmLWJjMzEtNDQ3NjM0MDUyMTc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIyVDEyMDc0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThhYjY1Nzk3NWI1Zjk4NTQ5OWJiYThiYzJiZmVjNmMwMDA2ZmFlNTNiYzZmYmI2NGIyYzA3ZmQ0NTc3MjI3ZmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gjbQmsp4a1cEMy63_fUzxSW2kDv3V3jW2sD1h-WTo3o)
> 
> https://www.awesomescreenshot.com/video/37901107?key=df9314644f6685c36d16a680631c17b2


Moved draft storage action inside the catch block of createAdCampaign request. This should fix the issue with not storing adds when wallet errors occur. 

Also fixed undefinde dates for prompts inside entry creation card (happened because of wrong merge conflict resolving back then)

Regarding images, it's intended to not store images in local storage for prompts/entries/cards, and is noted in PR description above FYI. 

 
<img width="1570" alt="Screenshot 2025-03-22 at 14 07 26" src="https://github.com/user-attachments/assets/bebe703a-3a28-4dd3-a248-eb55fe924a7f" />
<img width="943" alt="Screenshot 2025-03-22 at 14 07 10" src="https://github.com/user-attachments/assets/8b56c75f-444d-4fe6-a1fd-70327047e863" />

sana7044 commented on src/components/Admin/EntryCard.vue:
```diff
@@ -191,12 +251,22 @@ const promptOptions = computed(
     entry.description = props.description
     entry.image = props.image
-    entry.prompt = { label: `${props.prompt.date || props.prompt.publicationDate} – ${props.prompt.title}`, value: props.prompt.id }
+    entry.prompt = { label: `${props.prompt.date} – ${props.prompt.title}`, value: props.prompt.date }
```
### Entry edit time issue
![Screenshot from 2025-03-24 13-49-18](https://github.com/user-attachments/assets/cf2a7142-de19-45e4-93c0-7ffb9289cb1d)
use entry.prompt = { label: `${props.prompt.date || props.prompt.publicationDate} – ${props.prompt.title}`, value: props.prompt.id } this type

sana7044 commented on src/components/Admin/PromptCard.vue:
```diff
@@ -403,7 +415,19 @@ async function onProceedDepositFundDialog() {
+    prompt.showcase = parsedPrompt?.showcase
+    prompt.title = parsedPrompt?.title
+    prompt.id = parsedPrompt?.id
+
```
Please ensure that you can save the draft prompt with the publication date and end date . Also, reset the time when removing it
![Screenshot from 2025-03-24 13-59-01](https://github.com/user-attachments/assets/96389f88-40ee-43a3-b3f3-0c0fb888eda5)

sana7044 commented on src/components/Admin/EntryCard.vue:
```diff
@@ -261,8 +331,7 @@ async function onSubmit() {
   }
-  const date = props.id ? props.prompt.date || props.prompt.publicationDate : entry.prompt.date
-  entry.slug = `/${date.replace(/\-/g, '/')}/${entry.title.toLowerCase().replace(/[^0-9a-z]+/g, '-')}`
+  entry.slug = `/${entry.prompt.value.replace(/\-/g, '/')}/${entry.title.toLowerCase().replace(/[^0-9a-z]+/g, '-')}`
```
#### The entry slug is not working at view time
I think you can use the following code to generate the slug correctly, based on this type:
```
const date = props.id ? props.prompt.date || props.prompt.publicationDate : entry.prompt.date
  entry.slug = `/${date.replace(/\-/g, '/')}/${entry.title.toLowerCase().replace(/[^0-9a-z]+/g, '-')}`
```

sana7044 reviewed (changes requested):
@marclupanc, I’ve reviewed it and found a few minor issues. Everything else looks good to go!

marclupanc committed:
feat:#719 - Draft posts. WIP. Testing in staging env

marclupanc committed:
feat:#719 - conflicts resolving

marclupanc committed:
feat:#719 - Draft posts fixes
